pandas/core/base.py

```diff
@@ -990,13 +991,73 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
 
-        # FIXME in GH7447
-        # needs coercion on the key (DatetimeIndex does alreay)
-        # needs tests/doc-string
-        return self.values.searchsorted(key, side=side)
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
```

Comment:
ok, just about to merge, then I saw you changed *some* of the ``v`` -> ``value``. (you missed the datetimelikes). 

So originally we had datetimelikes used ``key`` (and I had you change it to ``v``). While ``Series`` used ``v``.

So we have to break something.

Let's just keep the rename to ``value`` (and fix the datetimelikes). Move the whatsnew note to ``API Changes`` (and note that the argument name changed). I seriously doubt *anyone* is actually passing as a kw arg (the first arg).